### ISSUE TITLE
bug: The functions "h3_exact_edge_length_m" and "h3_exact_edge_length_km" may exhibit cross-platform inconsistencies in floating-point values.

### ISSUE THREAD

**Issue by ct20000901** [datafuselabs/databend#12510]
### Search before asking

- [X] I had searched in the [issues](https://github.com/datafuselabs/databend/issues) and found no similar issues.

### Version

<img width="436" alt="image" src="https://github.com/datafuselabs/databend/assets/52696678/c94da20f-d981-4034-8f53-25658551e1f4">
<img width="289" alt="image" src="https://github.com/datafuselabs/databend/assets/52696678/612ddca6-9668-4a48-9ae0-ab463bdb8564">

### What's Wrong?

<img width="577" alt="image" src="https://github.com/datafuselabs/databend/assets/52696678/6ff9cbde-8eb4-4e9f-aa69-533a1472f56a">
<img width="490" alt="image" src="https://github.com/datafuselabs/databend/assets/52696678/c75e830b-f38a-48fc-8dc6-69741732b0c0">

however, the expected output is:
<img width="741" alt="image" src="https://github.com/datafuselabs/databend/assets/52696678/2f1be664-d145-483e-bc4a-37fde1d74cd7">

This problem causes my unit test to fail.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

**Comment by ct20000901** [comment]
It appears that all functions related to "geo_h3" may experience similar issues with cross-platform floating-point inconsistencies. cc @ariesdevil.  Thank you very much, and I appreciate your assistance.

**Comment by ariesdevil** [comment]
@ct20000901 The core issue in this situation is caused by the bit width provided by different float processor registers. This was addressed by IBM when they provided an 'Extended Precision' extension in their standard. This extension allows the float bit width to be expanded to 80-bit. For instance, the Motorola 88k architecture employs this extension, as does Apple (iOS).

The specific differences in parameters are explained in the link below, which discusses the relationship and specifics of processor register precision:
https://en.wikipedia.org/wiki/Processor_register

In other words, this is a computer problem.

**Comment by ct20000901** [comment]
> @ct20000901 The core issue in this situation is caused by the bit width provided by different float processor registers. This was addressed by IBM when they provided an 'Extended Precision' extension in their standard. This extension allows the float bit width to be expanded to 80-bit. For instance, the Motorola 88k architecture employs this extension, as does Apple (iOS).
> 
> The specific differences in parameters are explained in the link below, which discusses the relationship and specifics of processor register precision: https://en.wikipedia.org/wiki/Processor_register
> 
> In other words, this is a computer problem.

I understand this, but it may cause unit tests to fail on the macOS platform. Therefore, it might be necessary to modify the test cases or optimize this part to address the issue?

**Comment by ariesdevil** [comment]
@ct20000901 I create a new issue for this, we may need to refactor all float-point return value tests like this one.
